### PR TITLE
[ESP32] Support Thread and Wi-Fi network revert configuration

### PR DESCRIPTION
**Pull request by DejinChen**
### Problem
Platform ESP32 do not support Wi-Fi reverting network nor Thread reverting network to disconnect state.

### Change overview
- Removed check for KVS value point to support Thread reverting network to disconnect state
- Added RevertConfiguration for Wi-Fi

### Testing
RevertConfiguration for Wi-Fi and Thread network on ESP32-C6.

### CODE REVIEW

**Comment by DejinChen** [comment]
@chshu @dhrishi PTAL.

File: src/platform/ESP32/NetworkCommissioningDriver.cpp [hunk 1]
```diff
@@ -95,39 +95,31 @@ CHIP_ERROR GetConfiguredNetwork(Network & network)
...
 {
-    CHIP_ERROR err;
-    size_t ssidLen        = 0;
-    size_t credentialsLen = 0;
-
-    err = PersistedStorage::KeyValueStoreMgr().Get(kWiFiCredentialsKeyName, mSavedNetwork.credentials,
-                                                   sizeof(mSavedNetwork.credentials), &credentialsLen);
-    if (err == CHIP_ERROR_NOT_FOUND)
+    wifi_config_t stationConfig;
+    if (esp_wifi_get_config(WIFI_IF_STA, &stationConfig) == ERR_OK && stationConfig.sta.ssid[0] != 0)
```
Comment (wqx6) [inline src/platform/ESP32/NetworkCommissioningDriver.cpp:99]:
```suggestion
    if (esp_wifi_get_config(WIFI_IF_STA, &stationConfig) == ESP_OK && stationConfig.sta.ssid[0] != 0)
```

File: src/platform/ESP32/KeyValueStoreManagerImpl.cpp [hunk 2]
```diff
@@ -68,8 +68,6 @@ KeyValueStoreManagerImpl KeyValueStoreManagerImpl::sInstance;
 CHIP_ERROR KeyValueStoreManagerImpl::_Get(const char * key, void * value, size_t value_size, size_t * read_bytes_size,
                                           size_t offset_bytes)
 {
-    VerifyOrReturnError(value, CHIP_ERROR_INVALID_ARGUMENT);
```
Comment (wqx6) [inline src/platform/ESP32/KeyValueStoreManagerImpl.cpp:71]: Please add a comment about why to remove this check.

Comment (DejinChen) [reply src/platform/ESP32/KeyValueStoreManagerImpl.cpp:71]: done

**Review by wqx6 — APPROVED** [review]

**Review by chshu — APPROVED** [review]

**Review by yufengwangca — APPROVED** [review]

**Pull request merged**